service/matching/ratelimit_manager.go

```diff
@@ -154,30 +235,34 @@ func (r *rateLimitManager) trySetRPSFromUserDataLocked() {
 	}
```

Comment:
if config == nil, don't we have to set `r.apiConfigRPS` to nil? but if we just remove this `if config == nil` check, the next few lines will handle that properly